tests/thread_priority_inversion/main.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2017 Technische Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup tests
+ * @{
+ *
+ * @file
+ * @brief       Thread test application for priority inversion problem
+ *
+ * @author      Thomas Geithner <thomas.geithner@dai-labor.de>
+ *
+ * @}
+ */
+
+
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+
+#include "thread.h"
+#include "mutex.h"
+#include "xtimer.h"
+
+mutex_t res_mtx;
+
+char stack_high[THREAD_STACKSIZE_MAIN];
```

Comment:
IMHO `THREAD_STACKSIZE_DEFAULT` will be sufficient, too and will lower required memory considerably.